Source/Core/VideoCommon/HiresTextures.cpp

```diff
@@ -139,8 +139,11 @@ void HiresTexture::Prefetch()
 {
 	Common::SetCurrentThreadName("Prefetcher");
 
-	size_t size_sum = 0;
-	size_t max_mem = MemPhysical() / 2;
+	size_t size_sum = 0;	
+	size_t sys_mem = MemPhysical();
+	size_t recommended_min_mem = 2 * (1024.0 * 1024.0);
+	// keep 2GB memory for system stability if system RAM is 4GB+ - use half of memory in other cases
+	size_t max_mem = (sys_mem - recommended_min_mem < recommended_min_mem) ? (sys_mem / 2) : (sys_mem - recommended_min_mem);
```

Comment:
Actually good point. I'll push fix later + try to deal with Windows compiler warnings (normally use Mac). I think that line 144 should be 2.0 * (1024.0 * 1024.0) - now it's double to size_t conversion warning.